src/dex/cables/cables.ts

```diff
@@ -0,0 +1,1006 @@
+  CABLES_API_TOKENS_POLLING_INTERVAL_MS,
+  CABLES_API_URL,
+  CABLES_BLACKLIST_CACHE_KEY,
+  CABLES_BLACKLIST_CACHES_TTL_S,
+  CABLES_ERRORS_CACHE_KEY,
+  CABLES_FIRM_QUOTE_TIMEOUT_MS,
+  CABLES_GAS_COST,
+  CABLES_PAIRS_CACHES_TTL_S,
+  CABLES_PRICES_CACHES_TTL_S,
+  CABLES_RESTRICT_CHECK_INTERVAL_MS,
+  CABLES_RESTRICT_COUNT_THRESHOLD,
+  CABLES_RESTRICT_TTL_S,
+  CABLES_RESTRICTED_CACHE_KEY,
+  CABLES_TOKENS_CACHES_TTL_S,
+} from './constants';
+import { CablesRateFetcher } from './rate-fetcher';
+import {
+  CablesData,
+  CablesRFQResponse,
+  RestrictData,
+  SlippageError,
+} from './types';
+import mainnetRFQAbi from '../../abi/cables/CablesMainnetRFQ.json';
+import { Interface } from 'ethers/lib/utils';
+import BigNumber from 'bignumber.js';
+import { ethers } from 'ethers';
+import { BI_MAX_UINT256 } from '../../bigint-constants';
+import _ from 'lodash';
+
+export class Cables extends SimpleExchange implements IDex<any> {
+  public static dexKeysWithNetwork: { key: string; networks: Network[] }[] =
+    getDexKeysWithNetwork(CablesConfig);
+
+  readonly isStatePollingDex = true;
+
+  private rateFetcher: CablesRateFetcher;
+
+  logger: Logger;
+  private tokensMap: { [address: string]: Token } = {};
+
+  hasConstantPriceLargeAmounts: boolean = false;
+
+  constructor(
+    readonly network: Network,
+    readonly dexKey: string,
+    readonly dexHelper: IDexHelper,
+    readonly mainnetRFQAddress: string = CablesConfig['Cables'][network]
+      .mainnetRFQAddress,
+    protected rfqInterface = new Interface(mainnetRFQAbi),
+  ) {
+    super(dexHelper, dexKey);
+    this.logger = dexHelper.getLogger(dexKey);
+
+    this.rateFetcher = new CablesRateFetcher(
+      this.dexHelper,
+      this.dexKey,
+      this.network,
+      this.logger,
+      {
+        rateConfig: {
+          pairsReqParams: {
+            url: CABLES_API_URL + '/pairs',
+          },
+          pricesReqParams: {
+            url: CABLES_API_URL + '/prices',
+          },
+          blacklistReqParams: {
+            url: CABLES_API_URL + '/blacklist',
+          },
+          tokensReqParams: {
+            url: CABLES_API_URL + '/tokens',
+          },
+
+          pricesIntervalMs: CABLES_API_PRICES_POLLING_INTERVAL_MS,
+          pricesCacheTTLSecs: CABLES_PRICES_CACHES_TTL_S,
+          pricesCacheKey: 'prices',
+
+          pairsIntervalMs: CABLES_API_PAIRS_POLLING_INTERVAL_MS,
+          pairsCacheTTLSecs: CABLES_PAIRS_CACHES_TTL_S,
+          pairsCacheKey: 'pairs',
+
+          tokensIntervalMs: CABLES_API_TOKENS_POLLING_INTERVAL_MS,
+          tokensCacheTTLSecs: CABLES_TOKENS_CACHES_TTL_S,
+          tokensCacheKey: 'tokens',
+
+          blacklistIntervalMs: CABLES_API_BLACKLIST_POLLING_INTERVAL_MS,
+          blacklistCacheTTLSecs: CABLES_BLACKLIST_CACHES_TTL_S,
+          blacklistCacheKey: CABLES_BLACKLIST_CACHE_KEY,
+
+          tokensHandleResponseCallback: async () => {
+            await this.setTokensMap();
+          },
```

Comment:
Why do we need this if it's used only on init service?

Reply:
removed tokensHandleResponseCallback